### DISCUSSION TITLE
In a for loop on xonsh's REPL, how can I detect that Ctrl-C was issued?

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#5371] [Q&A] [answered]
Sometimes I have to write in the xonsh REPL a for-loop of long running commands . For example:

```xsh
For d in data_set_list:
   work_on_dataset @(d)
```

When I press Ctrl-c during the long running command `work_on_dataset @(d)` ,this command aborts as expected, but xonsh continues with the next iteration of `d`. How can I check after returning from `work_on_dataset @(d)` whether the command was aborted or not (in order to abort the loop then).

#### MWE

Here an minimal working example everybody can try:

```xsh
for i in range(5):
    print(i)
    sleep 10
```

When I Ctrl-c out of the `sleep`, xonsh continues with the next iteration.

#### What I've tried

* I tried using `rtn =  !(sleep 5)`, but then I cannot abort anymore at all!

* Then I experimented with `rtn = $(sleep 5)` and this aborts, but `rtn` is always the same afterwards, so I cannot use it to determine whether `sleep 5` was aborted.

**Comment by anki-code** [comment]
Hey! I know this case and I agree it's really annoying. 
I'm working on things around processes now and I traced this behavior. Here is the code:

https://github.com/xonsh/xonsh/blob/26c6e119e259ee6eac990233f8b0710cb67ea6b2/xonsh/jobs.py#L269-L288

When you run `sleep 10` the system process go to wait and we stuck on `os.waitpid`. After pressing `Ctrl+C` (sending SIGINT) process get the SIGINT signal and we catch that signal appeared using `WIFSIGNALED` (line 281) but have no actions about stopping the execution entirely and we continue execution.
It would be cool to considering the cases around this (i.e. can we really stop the execution of the whole code?) and stopping the execution carefully here. A bunch of [helpful tools](https://github.com/xonsh/xonsh/pull/5361#issuecomment-2078826181) is here. PR is welcome!

**Comment by anki-code** [comment, marked as answer]
Fixed in #5380. Set `$RAISE_SUBPROC_ERROR = True` before loop.
